content/docs/user-guide/setup-google-drive-remote.md

```diff
@@ -1,9 +1,122 @@
+command that needs it (e.g. `dvc pull`, `dvc fetch`, `dvc push`). For example:
+
+```dvc
+$ dvc remote add -d gdremote gdrive://0AIac4JZqHhKmUk9PDA/dvcstore
+$ dvc push
+
+Go to the following link in your browser:
+
+    https://accounts.google.com/o/oauth2/auth # ... copy this link
+
+Enter verification code: # <- enter resulting code
+```
+
+See the [Authorization](#authorization) section for more details on connecting
+DVC with Google Drive.
+
+DVC uses the Google Drive API to synchronize your <abbr>DVC project</abbr> data
+with this type of remote storage, so it's subject to certain usage limits and
+quotas, which by default are shared with other GDrive remote storage users. For
+heavy use, it's highly recommended to
+[connect using a custom Google Cloud project](#connect-using-a-custom-google-cloud-project),
+which puts you in control of these limits.
+
+Having your own GC project, it's also possible to
+[use a service account](#using-service-accounts) for automating tasks that need
+to establish GDrive remote connections (e.g. CI/CD).
+
+## URL formats to specify folder location
+
+There's a few alternatives to construct a GDrive remote URL for different uses,
+such as a folder or subfolder in root, shared folders not owned by your account,
+etc. The URL is formed with a _base_, and an optional _path_ to an **existing**
+folder i.e. `gdrive://<base>/path/to/folder`. The base can be one of:
+
+1. _Folder ID_ - unique identifier for each Google Drive folder
+
+   To obtain it, navigate to that folder in your web browser, and find it in the
+   address bar. For example, for
+   `https://drive.google.com/drive/folders/0AIac4JZqHhKmUk9PDA`:
+
+   ```dvc
+   $ dvc remote add gdfolder gdrive://0AIac4JZqHhKmUk9PDA
+   $ dvc remote add gdsubfolder \
+                  gdrive://0AIac4JZqHhKmUk9PDA/Data/text
+   ```
+
+   Note that [shared folders](https://support.google.com/drive/answer/7166529)
+   and [shared drives](https://support.google.com/a/users/answer/9310351) can
+   only be referenced this way.
+
+2. `root` - indicates your topmost Google Drive directory.
+
+   ⚠️ Only suitable for personal use, as sharing a remote configured this way
+   would cause DVC to try synchronizing data to/from different Google Drives for
+   every user.
+
+   ```dvc
+   $ dvc remote add mygdfolder gdrive://root/dvcstore
+   ```
+
+   > Although valid, we don't recommend using just `gdrive://root`, as the
+   > account's Google Drive home ("My Drive") could fill with cache directories,
+   > complicating managing non-DVC files and folders.
+
+3. `appDataFolder` -
+   [special hidden folder](https://developers.google.com/drive/api/v2/appdata)
+   (unique per user) meant to store application-specific data. This is a good
+   choice to prevent accidentally deleting remote storage data from the Google
+   Drive web UI.
+
+   ⚠️ Only suitable for personal use.
+
+   ```dvc
+   $ dvc remote add gdappata gdrive://appDataFolder
```

Comment:
Should it be lower case, something like `gdrive://appdata` instead, maybe? I don't think we use cammelCase much in these kind of literals inside DVC in general. Not "pythonic", I think. Obviously not a huge deal.

@shcheklein @maxhora

Reply:
`appDataFolder` is a predefined Google Drive specific name, it's not something we control. Same as `root`.

Reply:
Yes, makes sense. But the remote URL is an internal thing, they could be mapped in the code to what Google expects.

Reply:
this is a good point. If we support root and appData I would still keep names as is - to be more explicit. It might be though for simplicity we can just decide to support only URLs like: 

`gdrive://<folder-id>` in the future to make less error prone.

at least it feels like we should recommend this approach as much as possible.

Reply:
Agree.

> we should recommend this approach

Added "recommended" in folder id, and both other ways have warnings about being only good for personal use, so I think it's clear now.